### DISCUSSION TITLE
LXML Options (indent size)

### DISCUSSION THREAD

**Discussion by jtsage** [StjerneIdioten/I3D-Blender-Addon#145] [Ideas & Feature Requests]
I know, this is just picky, and yes, vscode lets me reflow the document.  But if someday we could have a subset of options for lxml exposed - namely indent size - 

https://stackoverflow.com/questions/1238988/changing-the-default-indentation-of-etree-tostring-in-lxml

I do prefer 4 spaces to the default 2.  Well, really, I prefer tabs like a not-crazy person, but most of the stuff I pull from the modhub is 4 spaces, so I am willing to follow suit.

**Comment by kbrandwijk** [comment]
Spaces FTW! https://hoffa.medium.com/400-000-github-repositories-1-billion-files-14-terabytes-of-code-spaces-or-tabs-7cfe0b5dd7fd#.fswbmzt2l

**Comment by StjerneIdioten** [comment]
This have got to be one of the most peculiar requests I've gotten 😆 But actually as it is now, lxml isn't needed anymore. The current implementation I am working with doesn't use it after python has finally given the built in library it deserves 😉 It now has the function `indent()` that can be used to pretty print the xml, but I suppose the option could still be exposed for customisation 😆

**Reply by jtsage** [reply]
LOL.  Hey, I even refrained from trying to start a flame war with Kim over spaces vs. tabs.  Even though it turns out the vast majority of people on github are wrong 😇   - that's excellent on the built in, didn't know they had done that, I'm going to have to take a look.

**Reply by StjerneIdioten** [reply]
Good, otherwise you would be outnumbered in here as well in that war 😉 

The [functionality](https://docs.python.org/3/library/xml.etree.elementtree.html#xml.etree.ElementTree.indent) was introduced in Python 3.9 and since that is the version Blender 2.93 ships with, it seemed fitting to go with just the built in library for now. I might backport some stuff to support older python versions at some point, but for now I will just focus on actually getting something which works for that version 😄
